app/src/main/java/ch/epfl/sdp/peakar/utils/OnSwipeTouchListener.java

```diff
@@ -0,0 +1,148 @@
+package ch.epfl.sdp.peakar.utils;
+
+import android.content.Context;
+import android.content.Intent;
+import android.util.DisplayMetrics;
+import android.view.GestureDetector;
+import android.view.MotionEvent;
+import android.view.View;
+
+import ch.epfl.sdp.peakar.camera.CameraActivity;
+import ch.epfl.sdp.peakar.gallery.GalleryActivity;
+import ch.epfl.sdp.peakar.general.SettingsActivity;
+import ch.epfl.sdp.peakar.map.MapActivity;
+import ch.epfl.sdp.peakar.social.SocialActivity;
+
+/**
+ * Detects left and right swipes across a view.
+ */
+public class OnSwipeTouchListener implements View.OnTouchListener {
+
+    private final GestureDetector gestureDetector;
+    private final Context context;
+
+    /**
+     * Constructor of the OnSwipeListener, creates the GestureListener
+     * @param context context of the application
+     */
+    public OnSwipeTouchListener(Context context) {
+        gestureDetector = new GestureDetector(context, new GestureListener());
+        this.context = context;
+    }
+
+    /**
+     * Starts the intent for the left swipe
+     * @param intent intent to be started
+     */
+    public void onSwipeLeft(Intent intent) {
+        intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
+        context.startActivity(intent);
+    }
+
+    /**
+     * Starts the intent for the right swipe
+     * @param intent intent to be started
+     */
+    public void onSwipeRight(Intent intent) {
+        intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
+        context.startActivity(intent);
+    }
+
+    /**
+     * Triggers callback for swipe gestures detection
+     *
+     * @param v view
+     * @param event current motion event
+     * @return True if motion has been consumed, else false
+     */
+    public boolean onTouch(View v, MotionEvent event) {
```

Comment:
Cirrus complaining about preform click should be called when click is detected

Reply:
Added @SuppressLint("ClickableViewAccessibility"). Since we only want to detect swipes, we do not need to call perform click